static/app/views/issueList/draggableTabBar.tsx

```diff
@@ -78,12 +79,18 @@ export function DraggableTabBar({
   onDelete,
   onDiscard,
   onDuplicate,
-  onRename,
+  onTabRenamed,
   onSave,
   onDiscardTempView,
   onSaveTempView,
 }: DraggableTabBarProps) {
   const [selectedTabKey, setSelectedTabKey] = useState<Key>(tabs[0].key);
+  const [isEditingTabTitle, setIsEditingTabTitle] = useState<{
+    isEditing: boolean;
```

Comment:
Hmm I don't think you need this `isEditing` state - could just have `editingTabKey: string | null` where null means nothing is being edited.

One other way you could simplify things (if you wanted, not required since this works well enough), is put this state as low as possible. You could extract the tab item into its own component and just have `const [isEditing, setIsEditing] = useState(false)`. Your new component would look something like:

```tsx
return (
  <Tabs>
    <DraggableTabList>
      {tabs.map(tab => <TabItem {...} />)}
    </DraggableTabList>
  </Tabs>
)

// in another file

function TabItem(...) {
  const [isEditing, setIsEditing] = useState(false)

  return (
   <DraggableTabList.Item>
     {...}
   </DraggableTabList.Item>
  )
}
```

It's always preferred to place the state as low as you can because it reduces rerenders and generally makes things simpler. The one downside is that you _could_ technically have multiple tabs editing at once, but I don't think that is actually possible since you exit editing on blur.

Reply:
Good point about the state! I also agree that the editing state makes more sense to apply to just the TabItem, but for the sake of getting this out to internal as soon as possible, I'll leave that out of this PR for now.